File: code/plot/trim_output.py
import os
import pandas as pd

# IO
fp = '/Users/mlandis/projects/gh_vib_div/'
in_fp = fp + 'output_raw/'
out_fp = fp + 'output/'

# get files
files = []
for r,d,f in os.walk(in_fp):
    for fn in f:
        files.append(fn)
files = [ x for x in files if 'out' in x ]

# process settings
n_it = 2001
it = [ x*50 for x in range(0,n_it) ]
#print(it)

# process files
#files = [ files[0] ]
#files = [ x for x in files if 'model' in x ]
for fn in files:
    in_fn = in_fp + fn
    df = pd.read_csv(in_fn, sep='\t', dtype=str)
    print(in_fn)
    #n_row = df.shape[0]
    print(df)
    df = df[-n_it:]
    df = df.assign(Iteration=it)
    print(df)
    print(df)
    df.to_csv(out_fp + fn, sep='\t', index=False)

File: code/trim_output.R
in_fp = "/Users/mlandis/projects/gh_vib_div/output_raw/"
out_fp = "/Users/mlandis/projects/gh_vib_div/output/"
files = list.files(path=fp)

n_keep = 2001

#files = files[1]
for (fn in files) {
    #print(fn)
    x = read.table(paste0(in_fp,fn), sep="\t", header=T, stringsAsFactors=F)
    it = unique(x$Iteration)
    n_it = length(it)
    x = x[(n_it-n_keep+1):n_it, ]
    x$Iteration = seq(0,100000,by=50)
    #print(x)
    write.table( x, paste0(out_fp,fn), sep="\t", quote=F)
}